spec/operatoroverloading.dd

```diff
@@ -514,6 +515,28 @@ struct S
         version of $(D opEquals).  Otherwise, inequalities like $(D a <= b)
         will behave inconsistently with equalities like $(D a == b).)
 
+$(SPEC_RUNNABLE_EXAMPLE_RUN
+---
+struct S
+{
+    int i, j;
+    int opCmp(ref const S s) const { return (i > s.i) - (i < s.i); } // ignore j
+}
+
+S a = {2, 3};
+S b = {2, 1};
+S c = {3, 0};
+assert(a < c);
+assert(a <= b);
+assert(!(a < b)); // opCmp ignores j
+assert(a != b);   // generated opEquals tests both i and j members
+---
+)
+
+$(BEST_PRACTICE Using `(i > s.i) - (i < s.i)` instead of `i - s.i` to
+compare integers avoids overflow/underflow.)
```

Comment:
```suggestion
compare integers avoids overflow.)
```

underflow happens with small floating points, going below `int.min` is officially still called overflow